File: testdata/jennies/rawtypes/struct_with_scalar_fields/PHPRawTypes/src/Basic/SomeStruct.php
<?php

namespace Grafana\Foundation\Basic;

/**
 * This
 * is
 * a
 * comment
 */
class SomeStruct implements \JsonSerializable
{
    /**
     * Anything can go in there.
     * Really, anything.
     * @var mixed
     */
    public $fieldAny;

    public bool $fieldBool;

    public string $fieldBytes;

    public string $fieldString;

    public string $fieldStringWithConstantValue;

    public float $fieldFloat32;

    public float $fieldFloat64;

    public int $fieldUint8;

    public int $fieldUint16;

    public int $fieldUint32;

    public int $fieldUint64;

    public int $fieldInt8;

    public int $fieldInt16;

    public int $fieldInt32;

    public int $fieldInt64;

    /**
     * @param mixed|null $fieldAny
     * @param bool|null $fieldBool
     * @param string|null $fieldBytes
     * @param string|null $fieldString
     * @param float|null $fieldFloat32
     * @param float|null $fieldFloat64
     * @param int|null $fieldUint8
     * @param int|null $fieldUint16
     * @param int|null $fieldUint32
     * @param int|null $fieldUint64
     * @param int|null $fieldInt8
     * @param int|null $fieldInt16
     * @param int|null $fieldInt32
     * @param int|null $fieldInt64
     */
    public function __construct( $fieldAny = null, ?bool $fieldBool = null, ?string $fieldBytes = null, ?string $fieldString = null, ?float $fieldFloat32 = null, ?float $fieldFloat64 = null, ?int $fieldUint8 = null, ?int $fieldUint16 = null, ?int $fieldUint32 = null, ?int $fieldUint64 = null, ?int $fieldInt8 = null, ?int $fieldInt16 = null, ?int $fieldInt32 = null, ?int $fieldInt64 = null)
    {
        $this->fieldAny = $fieldAny ?: null;
        $this->fieldBool = $fieldBool ?: false;
        $this->fieldBytes = $fieldBytes ?: "";
        $this->fieldString = $fieldString ?: "";
        $this->fieldStringWithConstantValue = "auto";
    
        $this->fieldFloat32 = $fieldFloat32 ?: 0;
        $this->fieldFloat64 = $fieldFloat64 ?: 0;
        $this->fieldUint8 = $fieldUint8 ?: 0;
        $this->fieldUint16 = $fieldUint16 ?: 0;
        $this->fieldUint32 = $fieldUint32 ?: 0;
        $this->fieldUint64 = $fieldUint64 ?: 0;
        $this->fieldInt8 = $fieldInt8 ?: 0;
        $this->fieldInt16 = $fieldInt16 ?: 0;
        $this->fieldInt32 = $fieldInt32 ?: 0;
        $this->fieldInt64 = $fieldInt64 ?: 0;
    }

    /**
     * @param array<string, mixed> $inputData
     */
    public static function fromArray(array $inputData): self
    {
        /** @var array{FieldAny?: mixed, FieldBool?: bool, FieldBytes?: string, FieldString?: string, FieldStringWithConstantValue?: string, FieldFloat32?: float, FieldFloat64?: float, FieldUint8?: int, FieldUint16?: int, FieldUint32?: int, FieldUint64?: int, FieldInt8?: int, FieldInt16?: int, FieldInt32?: int, FieldInt64?: int} $inputData */
        $data = $inputData;
        return new self(
            fieldAny: $data["FieldAny"] ?? null,
            fieldBool: $data["FieldBool"] ?? null,
            fieldBytes: $data["FieldBytes"] ?? null,
            fieldString: $data["FieldString"] ?? null,
            fieldFloat32: $data["FieldFloat32"] ?? null,
            fieldFloat64: $data["FieldFloat64"] ?? null,
            fieldUint8: $data["FieldUint8"] ?? null,
            fieldUint16: $data["FieldUint16"] ?? null,
            fieldUint32: $data["FieldUint32"] ?? null,
            fieldUint64: $data["FieldUint64"] ?? null,
            fieldInt8: $data["FieldInt8"] ?? null,
            fieldInt16: $data["FieldInt16"] ?? null,
            fieldInt32: $data["FieldInt32"] ?? null,
            fieldInt64: $data["FieldInt64"] ?? null,
        );
    }

    /**
     * @return mixed
     */
    public function jsonSerialize(): mixed
    {
        $data = new \stdClass;
        $data->FieldAny = $this->fieldAny;
        $data->FieldBool = $this->fieldBool;
        $data->FieldBytes = $this->fieldBytes;
        $data->FieldString = $this->fieldString;
        $data->FieldStringWithConstantValue = $this->fieldStringWithConstantValue;
        $data->FieldFloat32 = $this->fieldFloat32;
        $data->FieldFloat64 = $this->fieldFloat64;
        $data->FieldUint8 = $this->fieldUint8;
        $data->FieldUint16 = $this->fieldUint16;
        $data->FieldUint32 = $this->fieldUint32;
        $data->FieldUint64 = $this->fieldUint64;
        $data->FieldInt8 = $this->fieldInt8;
        $data->FieldInt16 = $this->fieldInt16;
        $data->FieldInt32 = $this->fieldInt32;
        $data->FieldInt64 = $this->fieldInt64;
        return $data;
    }
}

File: testdata/jennies/rawtypes/struct_with_scalar_fields/JavaRawTypes/basic/SomeStruct.java
package basic;

import java.util.Objects;

// This
// is
// a
// comment
public class SomeStruct {
    // Anything can go in there.
    // Really, anything.
    public Object fieldAny;
    public Boolean fieldBool;
    public Byte fieldBytes;
    public String fieldString;
    public String fieldStringWithConstantValue;
    public Float fieldFloat32;
    public Double fieldFloat64;
    public Integer fieldUint8;
    public Short fieldUint16;
    public Integer fieldUint32;
    public Long fieldUint64;
    public Integer fieldInt8;
    public Short fieldInt16;
    public Integer fieldInt32;
    public Long fieldInt64;
    public SomeStruct() {
        this.fieldAny = new Object();
        this.fieldBool = false;
        this.fieldBytes = (byte) 0;
        this.fieldString = "";
        this.fieldStringWithConstantValue = "";
        this.fieldFloat32 = 0.0f;
        this.fieldFloat64 = 0.0;
        this.fieldUint8 = 0;
        this.fieldUint16 = 0;
        this.fieldUint32 = 0;
        this.fieldUint64 = 0L;
        this.fieldInt8 = 0;
        this.fieldInt16 = 0;
        this.fieldInt32 = 0;
        this.fieldInt64 = 0L;
    }
    public SomeStruct(Object fieldAny,Boolean fieldBool,Byte fieldBytes,String fieldString,String fieldStringWithConstantValue,Float fieldFloat32,Double fieldFloat64,Integer fieldUint8,Short fieldUint16,Integer fieldUint32,Long fieldUint64,Integer fieldInt8,Short fieldInt16,Integer fieldInt32,Long fieldInt64) {
        this.fieldAny = fieldAny;
        this.fieldBool = fieldBool;
        this.fieldBytes = fieldBytes;
        this.fieldString = fieldString;
        this.fieldStringWithConstantValue = fieldStringWithConstantValue;
        this.fieldFloat32 = fieldFloat32;
        this.fieldFloat64 = fieldFloat64;
        this.fieldUint8 = fieldUint8;
        this.fieldUint16 = fieldUint16;
        this.fieldUint32 = fieldUint32;
        this.fieldUint64 = fieldUint64;
        this.fieldInt8 = fieldInt8;
        this.fieldInt16 = fieldInt16;
        this.fieldInt32 = fieldInt32;
        this.fieldInt64 = fieldInt64;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof SomeStruct)) return false;
        SomeStruct o = (SomeStruct) other;
        if (!Objects.equals(this.fieldAny, o.fieldAny)) return false;
        if (!Objects.equals(this.fieldBool, o.fieldBool)) return false;
        if (!Objects.equals(this.fieldBytes, o.fieldBytes)) return false;
        if (!Objects.equals(this.fieldString, o.fieldString)) return false;
        if (!Objects.equals(this.fieldStringWithConstantValue, o.fieldStringWithConstantValue)) return false;
        if (!Objects.equals(this.fieldFloat32, o.fieldFloat32)) return false;
        if (!Objects.equals(this.fieldFloat64, o.fieldFloat64)) return false;
        if (!Objects.equals(this.fieldUint8, o.fieldUint8)) return false;
        if (!Objects.equals(this.fieldUint16, o.fieldUint16)) return false;
        if (!Objects.equals(this.fieldUint32, o.fieldUint32)) return false;
        if (!Objects.equals(this.fieldUint64, o.fieldUint64)) return false;
        if (!Objects.equals(this.fieldInt8, o.fieldInt8)) return false;
        if (!Objects.equals(this.fieldInt16, o.fieldInt16)) return false;
        if (!Objects.equals(this.fieldInt32, o.fieldInt32)) return false;
        if (!Objects.equals(this.fieldInt64, o.fieldInt64)) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.fieldAny, this.fieldBool, this.fieldBytes, this.fieldString, this.fieldStringWithConstantValue, this.fieldFloat32, this.fieldFloat64, this.fieldUint8, this.fieldUint16, this.fieldUint32, this.fieldUint64, this.fieldInt8, this.fieldInt16, this.fieldInt32, this.fieldInt64);
    }
}
